### ISSUE TITLE
Error dropping empty database

### ISSUE THREAD

**Issue by Gaff** [bwaldvogel/mongo-java-server#136]
Hi,

Since this change: https://github.com/bwaldvogel/mongo-java-server/commit/ffdedaa453f0232507ecde20469383988a1f2ccb#diff-469784d941b7732e6c1c0b22c49e64a5R251

I get an error with a simple no-op test:

```java
        server = new MongoServer(new MemoryBackend());
        InetSocketAddress serverAddress = server.bind();
        client = new MongoClient(new ServerAddress(serverAddress));

       MongoDatabase db = client.getDatabase("testdb");
       db.drop();
```

Here's the error:

```
java.lang.NullPointerException: null
	at de.bwaldvogel.mongo.backend.AbstractMongoBackend.dropDatabase(AbstractMongoBackend.java:313)
	at de.bwaldvogel.mongo.backend.AbstractMongoBackend.handleDropDatabase(AbstractMongoBackend.java:304)
	at de.bwaldvogel.mongo.backend.AbstractMongoBackend.handleCommand(AbstractMongoBackend.java:251)
	at de.bwaldvogel.mongo.wire.MongoDatabaseHandler.handleCommand(MongoDatabaseHandler.java:183)
 ...etc
```

This used to work because when `db.drop()` was called the `AbstractMongoBackend` would call `MongoDatabase db = resolveDatabase(databaseName);` - which had the side effect of creating the DB if it didn't already exist - before dropping it.

**Comment by Gaff** [comment]
At the very least it shouldn't be a NPE, because downstream this causes a very non-obvious stream error:

```
com.mongodb.MongoSocketReadException: Prematurely reached end of stream

	at com.mongodb.connection.SocketStream.read(SocketStream.java:88)
	at com.mongodb.connection.InternalStreamConnection.receiveResponseBuffers(InternalStreamConnection.java:491)
	at com.mongodb.connection.InternalStreamConnection.receiveMessage(InternalStreamConnection.java:221)
	at com.mongodb.connection.UsageTrackingInternalConnection.receiveMessage(UsageTrackingInternalConnection.java:102)
	at com.mongodb.connection.DefaultConnectionPool$PooledConnection.receiveMessage(DefaultConnectionPool.java:433)
	at com.mongodb.connection.CommandProtocol.execute(CommandProtocol.java:109)
   ... etc
```

**Comment by bwaldvogel** [comment]
Thanks for reporting this issue. Such a test case was missing.

**Comment by bwaldvogel** [comment]
Should be fixed with 22bfabd

**Comment by bwaldvogel** [comment]
Fixed in version `1.29.0`.
